doc/bmp/bmp.md

```diff
@@ -0,0 +1,373 @@
+
+
+## 1.2 CLI requirements
+- Use Config cli to enable bmp table population as granularity of table
+- Use show to query specific bmp table data
+
+## 1.3 Scalability and Default Values
+
+As a phase #1 scalability requirement, the proposed support is to have bgp neighbor table enabled by default, considering the device set like 7050qx (4G RAM), but it should be based on final evaluation, more details will be discussed in [4 Resource usage and Test plan](#4-resource-usage-and-test-plan)
+
+
+## 1.4 Warm Restart requirements
+No special handling for Warm restart support.
+
+# 2 Architecture Design
+
+The following are the architecture changes. 
+
+## 2.1 High-Level Architecture
+As above context mentioned, we need to bring up new container as bmp on SONiC, after change the relevant component architecture is as below:
+<img src="images/architecture_diagram.png" alt="Architecture Diagram" width="800">  
+
+- Add new bmp container, which has limited resource control. 
+
+- Update existing frr/bgpd daemon and enable bmp feature to collect bgp data.
+
+- BMP will support both config and CLI, which can enable database population per table type population by default and in runtime.
+
+- User could use StreamingTelemetry path to monitor bgp table via [The GNMI protobuf](#https://github.com/openconfig/gnmi/blob/5473f2ef722ee45c3f26eee3f4a44a7d827e3575/proto/gnmi/gnmi.proto#L309 )
+
+## 2.2 OpenBMP
+[OpenBMP](#https://www.openbmp.org/) is open source BGP monitor framework, which supports bgp collecting, aggregating, data persistence, as well as dashboard/monitor/alert, etc. For this project, we will fork this project with some internal code changes which populates the required data into Redis for our usage.
+
+## 2.3 BGP/FRR update
+Current BGP container in SONiC uses an implementation [sonic-frr](#https://github.com/sonic-net/sonic-frr/) forked from [FRR](#https://github.com/FRRouting/frr/), which supports BMP protocol from version frr/7.2
+
+
+### Config file update
+Below section needs to be added into /etc/frr/bgpd.conf, so that FRR could be able to find assigned collector endpoint and report bgp data via bmp channel.
+
+```
+! 
+  bmp mirror buffer-limit 4294967214 
+! 
+  bmp targets test 
+    bmp stats interval 1000 
+    bmp monitor ipv4 unicast pre-policy 
+    bmp monitor ipv6 unicast pre-policy 
+    bmp connect 127.0.0.1 port 5000 min-retry 1000 max-retry 2000
+! 
+```
+
+### Daemon parameter
+From version frr/7.2, bgpd daemon supports parameter to enable bmp functionality, we need to adjust BGP docker init relevant script and turn on bmp collection option.
+
+```
+/usr/lib/frr/bgpd -A 127.0.0.1 -M bmp 
+```
+
+## 2.4 Database Design
+
+BMP will continually populate existing redis table APPL_DB, just with different keys defined to cover the data so that functional requirement could be supported.
```

Comment:
Please consider pushing this state information into STATE_DB or new DB instance if that makes our life easy.

Reply:
Updated.